### PR TITLE
Pass cider-default-cljs-repl value to server's on-port-callback

### LINKED ISSUE #2336: New connection manager reads the wrong cider-default-cljs-repl
## Expected behavior

When I call `C-c M-J` the `cider-connect-sibling-cljs` should read the `cider-default-cljs-repl` from the code buffer's locals.

## Actual behavior
It always reads from the REPL buffer.

## Steps to reproduce the problem

Have a `.dir-locals.el` like this in your folder:

```elisp
((nil . ((cider-default-cljs-repl . figwheel-main)))
```

`C-c M-J` in a code buffer. You will always be prompted no matter what is the value of `cider-default-cljs-repl`.

### CIDER version information
`latest master`

The problems seems in  [here](https://github.com/clojure-emacs/cider/blob/master/cider.el#L947), the variable should be read from the code buffer but it is read from the REPL instead.

### PR DESCRIPTION
**Pull request by vspinu**
Fixes #2336

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -899,10 +899,12 @@ Prompt for the project and nREPL server command when DO-PROMPT is non-nil."
   (interactive "P")
   (let ((cider-jack-in-dependencies (append cider-jack-in-dependencies cider-jack-in-cljs-dependencies))
         (cider-jack-in-lein-plugins (append cider-jack-in-lein-plugins cider-jack-in-cljs-lein-plugins))
-        (cider-jack-in-nrepl-middlewares (append cider-jack-in-nrepl-middlewares cider-jack-in-cljs-nrepl-middlewares)))
+        (cider-jack-in-nrepl-middlewares (append cider-jack-in-nrepl-middlewares cider-jack-in-cljs-nrepl-middlewares))
+        (cljs-repl-type cider-default-cljs-repl))
     (cider--jack-in do-prompt
       (lambda (server-buffer)
-        (cider-connect-sibling-cljs server-buffer)))))
+        (let ((cider-default-cljs-repl cljs-repl-type))
```
Comment (bbatsov) [inline cider.el]: I'm not a big fan of dynamic binding, as it makes the code harder to read. Maybe we should pass this as a param?

File: cider.el [hunk 2]
```diff
@@ -944,8 +948,7 @@ Figwheel, etc), unless you've set `cider-default-cljs-repl'.  OTHER-REPL
 can also be a server buffer, in which case a new session with a REPL for
 that server is created."
   (interactive (list (cider-current-repl)))
-  (let ((cljs-repl-type (or cider-default-cljs-repl
-                            (cider-select-cljs-repl)))
+  (let ((cljs-repl-type (or cider-default-cljs-repl (cider-select-cljs-repl)))
```
Comment (bbatsov) [inline cider.el]: Or we can obtain this differently here.

File: cider.el (hunk 1 above)
Comment (vspinu) [reply cider.el]: I think we should do it anyhow. `cider-connect-siblings-cljs` accepts other-repl as first and the only argument. This is useless for interactive use. So I would say we make two argument with the first one being either `cljs-repl-type` (or maybe more generic params) which would ask on C-u for repl type. Then even with dir-local `default-cljs-repl`  set one can spark a different repl type with `C-u C-c M-s`.

Comment (bbatsov) [reply cider.el]: OK

**Pull request merged**